django-backend/fecfiler/f3x_summaries/models.py

```diff
@@ -368,6 +372,72 @@ class F3XSummary(SoftDeleteModel, CommitteeOwnedModel):
     created = models.DateTimeField(auto_now_add=True)
     updated = models.DateTimeField(auto_now=True)
 
+    def save(self, *args, **kwargs):
+        # Record if the is a create or update operation
+        create_action = self.created is None
+
+        with db_transaction.atomic():
+            super(F3XSummary, self).save(*args, **kwargs)
+
+            # Pull forward any loans with non-zero balances along with their
+            # loan guarantors
+            if create_action and self.coverage_through_date:
+                self.pull_forward_loans()
+
+    def pull_forward_loans(self):
+
+        previous_report = (
+            F3XSummary.objects.get_queryset()
+            .filter(
+                committee_account=self.committee_account,
+                coverage_through_date__lt=self.coverage_through_date,
+                deleted__isnull=True,
+            )
+            .order_by("coverage_through_date")
+            .last()
+        )
+        if previous_report:
+            loans_to_pull_forward = F3XSummary.objects.get(
+                id=previous_report.id
+            ).transaction_set.filter(
```

Comment:
I believe `previous_report` and `F3XSummary.objects.get(
                id=previous_report.id
            )` are equivalent.
 can this be `previous_report.transaction_set.filter(...`

Reply:
change made